posthog/settings/web.py

```diff
@@ -47,10 +47,11 @@
     "django_structlog.middlewares.RequestMiddleware",
     "django_structlog.middlewares.CeleryMiddleware",
     "django.middleware.security.SecurityMiddleware",
-    "posthog.middleware.AllowIPMiddleware",
+    "posthog.middleware.ShortCircuitMiddleware",
```

Comment:
Overall I'm not a big fan of this approach as we are silently short circuiting all the middlewares below this line using a condition in the middleware itself. 

`/decide` is pretty simple endpoint to audit but we might be missing an application behaviour that can depends on one of middlewares below.

Reply:
Yeah definitely. I want to just test my hypothesis here, if it does work I want to apply this to the capture endpoint as well so I'll spend a bit more time

Reply:
Actually after reviewing this further this is a pattern we already use in our codebase https://github.com/PostHog/posthog/blob/46faa8fab258b04cf4aa6fa27fbe39563bb998d2/posthog/health.py#L237-L253

Reply:
I'll put this live and see if it makes a difference. If it does I'll refactor all this into something more sensible